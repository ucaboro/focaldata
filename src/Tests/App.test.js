import React from "react";
import renderer from "react-test-renderer";
import Navbar from "../Components/navbar";
import SurveyCard from "../Components/surveyCard";
import PieChart from "../Components/charts/pieChart";
import RadarChart from "../Components/charts/radarChart";
import ChartCard from "../Components/chartCard";
import Main from "../Containers/main";
import Survey from "../Containers/survey";
import { BrowserRouter as Router } from "react-router-dom";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure, mount } from "enzyme";

import {
  assignRelevantIcon,
  splitLongString,
  clipExcessText
} from "../constants/utils";

configure({ adapter: new Adapter() });

const longText =
  "Testing a long string with more than 60 characters to check that it splits properly";
const shortText = "short text";

describe("App functions", () => {
  describe("Splitting long string", () => {
    it("should return short text", () => {
      let result = clipExcessText(shortText, 60);
      expect(result).toEqual(shortText);
    });
    it("should split long text with *", () => {
      let result = clipExcessText(longText, 60);
      expect(result).toContain("*");
    });
    it("should return splitted array", () => {
      let result = splitLongString(longText);
      expect(result).toHaveLength(2);
    });
  });
  const texts = [
    "brexit was postponed",
    "US elections",
    "care about environment",
    "text with no keywords"
  ];

  describe("Assigning icons to text", () => {
    it("should return door icon", () => {
      let result = assignRelevantIcon(texts[0]);
      expect(result.props.className).toEqual("fa fa-3x fa-door-open");
    });
    it("should return flag icon", () => {
      let result = assignRelevantIcon(texts[1]);
      expect(result.props.className).toEqual("fa fa-3x fa-flag-usa");
    });
    it("should return leaf icon", () => {
      let result = assignRelevantIcon(texts[2]);
      expect(result.props.className).toEqual("fa fa-3x fa-leaf");
    });
    it("should return question icon", () => {
      let result = assignRelevantIcon(texts[3]);
      expect(result.props.className).toEqual("fa fa-3x fa-question");
    });
  });
});

//SNAPSHOTS
describe("Snapshots", () => {
  describe("Components", () => {
    test("navbar renders", () => {
      const component = renderer.create(
        <Router>
          <Navbar />
        </Router>
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    test("surveyCard renders", () => {
      const component = renderer.create(
        <SurveyCard key={"1"} name={"title"} questions={"4"} />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    test("chartCard renders", () => {
      const component = renderer.create(<ChartCard />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    const data = {
      questionId: 1,
      questionTitle: "test title",
      answerOptions: [
        { answerOption: 1, text: "text", selectedByRespondents: 40 },
        { answerOption: 2, text: "text 2", selectedByRespondents: 60 }
      ]
    };

    describe("Charts", () => {
      test("RadarChart renders", () => {
        const wrapper = shallow(
          <RadarChart
            data={data.answerOptions}
            title={data.questionTitle}
            colors={["#35477d", "#6c5b7b", "#c06c84", "#FF6E5A", "#f67280"]}
          />,
          { lifecycleExperimental: true }
        );

        expect(wrapper).toMatchSnapshot();
      });
      test("PieChart renders", () => {
        const wrapper = shallow(
          <PieChart
            data={["1", "2"]}
            title={"data.questionTitle"}
            colors={["#35477d", "#6c5b7b", "#c06c84", "#FF6E5A", "#f67280"]}
          />,
          { lifecycleExperimental: true }
        );
        expect(wrapper).toMatchSnapshot();
      });
    });
  });

  describe("Containers", () => {
    test("Main page renders with children", () => {
      const wrapper = mount(
        <Router>
          <Main />
        </Router>
      );
      expect(wrapper).toMatchSnapshot();
    });
    test("Survey page renders with children", () => {
      const wrapper = mount(
        <Router>
          <Survey />
        </Router>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
