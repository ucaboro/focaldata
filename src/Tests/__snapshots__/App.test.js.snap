// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots Components Charts PieChart renders 1`] = `ShallowWrapper {}`;

exports[`Snapshots Components Charts RadarChart renders 1`] = `ShallowWrapper {}`;

exports[`Snapshots Components chartCard renders 1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  />
</div>
`;

exports[`Snapshots Components navbar renders 1`] = `
<nav
  className="sc-bwzfXH gbjnra navbar is-transparent"
>
  <div
    className="sc-htpNat hBjRQC navbar-brand"
  >
    <div
      className="navbar-item"
      onClick={[Function]}
    >
      <img
        alt="focaldata_logo"
        className="sc-bxivhb jVkVnH"
        src="logo.png"
      />
    </div>
    <div
      className="navbar-burger burger "
      onClick={[Function]}
    >
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
    </div>
  </div>
  <div
    className="sc-EHOje evioSs navbar-menu "
  >
    <div
      className="navbar-end"
    >
      <a
        className="sc-ifAKCX dXJwZw navbar-item"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="sc-ifAKCX dXJwZw navbar-item"
        href="https://www.focaldata.com/company/about-us"
      >
        About
      </a>
    </div>
  </div>
</nav>
`;

exports[`Snapshots Components surveyCard renders 1`] = `
<div
  className="sc-bdVaJa evmvjk card"
>
  <div
    className="card-content"
  >
    <div
      className="media"
    >
      <div
        className="media-left"
      >
        <span
          className="icon is-large"
        >
          <i
            className="fa fa-3x fa-question"
          />
        </span>
      </div>
      <div
        className="media-content"
      >
        <p
          className="title is-4"
        >
          title
        </p>
        <p
          className="subtitle is-6"
        >
          Questions: 
          4
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshots Containers Main page renders with children 1`] = `ReactWrapper {}`;

exports[`Snapshots Containers Survey page renders with children 1`] = `ReactWrapper {}`;
